### DISCUSSION TITLE
How to find enabled Site Features

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#1368] [Q&A] [answered]
I can enable a feature with:
`await context.Site.Features.EnableAsync(FeatureID);` 
But if I do it again I get "This Feature is already Activated in the scope"
I tried 
`if (!context.Site.Features.IsSiteFeatureActive(FeatureID))` That Copilot suggested. But there is no such method.

How do I check the existence?

Thank you

**Comment by jansenbe** [comment, marked as answer]
Ensure the features are loaded (e.g. `await context.Site.EnsurePropertiesAsync(p => p.Features);`) and then verify if the feature you need was there:

```csharp
// Was the communication site feature enabled?
var communicationSiteFeature = Guid.Parse("f39dad74-ea79-46ef-9ef7-fe2370754f6f");
var feature = context.Site.Features.AsRequested().FirstOrDefault(p => p.DefinitionId == communicationSiteFeature);
bool enabled = feature != null;
```
